samples/vSphereAutoRestartManager.py

```diff
@@ -0,0 +1,118 @@
+        acthosts = commaList.split(",")
+        allhosts = self.get_hosts(connection)
+
+        for h in allhosts:
+            for a in acthosts:
+                if h.name == a:
+                    self.enable_autorestart(h, defstartdelay)
+                else:
+                    print "The host cant be found" + a
+
+    def enable_autorestart(self, host, defstartdelay):
+        print "Enabling Auto Restart for Host"
+        print "The Selected Host is \n"+host.name
+        print "Setting the Selected Host default AutoStartManager"
+        hostDefSettings = vim.host.AutoStartManager.SystemDefaults()
+        hostDefSettings.enabled = True
+        hostDefSettings.startDelay = int(defstartdelay)
+        print"virtual machines and applying Auto Start settings"
+        order = 1
+        for vhost in host.vm:
+            spec = host.configManager.autoStartManager.config
+            spec.defaults = hostDefSettings
+            auto_power_info = vim.host.AutoStartManager.AutoPowerInfo()
+            auto_power_info.key = vhost
+            print "The VM   is updated if On"+vhost.name
+            print "VM Status is " + vhost.runtime.powerState
+            if vhost.runtime.powerState == "poweredOff":
+                auto_power_info.startAction = 'None'
+                auto_power_info.waitForHeartbeat = 'no'
+                auto_power_info.startDelay = -1
+                auto_power_info.startOrder = -1
+                auto_power_info.stopAction = 'None'
+                auto_power_info.stopDelay = -1
+            elif vhost.runtime.powerState == "poweredOn":
+                # note use of constant instead of string
+                auto_power_info.startAction = 'powerOn'
+                auto_power_info.startDelay = -1
+                auto_power_info.startOrder = -1
+                auto_power_info.stopAction = 'None'
+                auto_power_info.stopDelay = -1
+                auto_power_info.waitForHeartbeat = 'no'
+                spec.powerInfo = [auto_power_info]
+                order = order + 1
+                print "Apply Setting to Host"
+                host.configManager.autoStartManager.ReconfigureAutostart(spec)
+
+# MAIN
+    def main(self):
+        print "Starting"
+        print "Getting Config"
+        print "Connecting to vSphere"
+        connection = self.get_connection(
+            self.params.ipadd, self.params.user, self.params.password)
+
+        if self.params.listallhosts is True:
+            vSpherehosts = self.get_hosts(connection)
+            print "All the Hosts Attached are :\n"
+            for hosts in vSpherehosts:
+                print "\n"+hosts.name
+
+        if self.params.actionhosts is not None:
+            self.action_hosts(
+                self.params.actionhosts, connection, self.params.defstartdelay)
+
+
+if __name__ == "__main__":
+    vsphere = vSphereTool()
+    vsphere.add_param("-ip", "--ipadd", help="Vsphere ESXI ip address",
```

Comment:
Maybe Im missing something here but since vSphereTool no longer extends that cli class you were using before how is add_param part of the class?